crud/count.lua

```diff
@@ -0,0 +1,195 @@
+local checks = require('checks')
+local errors = require('errors')
+local vshard = require('vshard')
+local fiber = require('fiber')
+
+local call = require('crud.common.call')
+local utils = require('crud.common.utils')
+local filters = require('crud.compare.filters')
+local count_plan = require('crud.common.base_plan')
+local dev_checks = require('crud.common.dev_checks')
+
+local compare_conditions = require('crud.compare.conditions')
+
+local CountError = errors.new_class('CountError', {capture_stack = false})
+
+local COUNT_FUNC_NAME = '_crud.count_on_storage'
+local DEFAULT_COUNT_TO_YIELD = 100
+
+local count = {}
+
+local function count_on_storage(space_name, index_id, conditions, opts)
+    dev_checks('string', 'number', '?table', {
+        scan_value = 'table',
+        tarantool_iter = 'number',
+        count_to_yield = '?number',
+        scan_condition_num = '?number',
+    })
+
+    opts = opts or {}
+
+    if opts.count_to_yield == 0 then
+        return {}
+    end
```

Comment:
This check was removed because we have suck check on router:

```lua
if opts.count_to_yield ~= nil and opts.count_to_yield < 1 then
     return nil, CountError:new("count_to_yield should be > 0")
end
```